Denise Escobar -Computer Architecture

DenEscobar opened the pull request.

DenEscobar committed:
Working on the constructor for cpu.py

DenEscobar committed:
finished constructor of cpu.py and added ram functions

DenEscobar committed:
implemented run method and fixed errors in constructor and ram methods...
-Running the tests gives no erros but also no results

kyle-kauzlarich commented on ls8/cpu.py:
```diff
@@ -7,6 +7,9 @@ class CPU:
 
     def __init__(self):
         """Construct a new CPU."""
+        self.ram = [0] *256
```
why are you using  [0] *256 while using [0] * 8 in the next line?

DenEscobar replied:
I thought the ram and registers were two separate things? So [0] * 8 is for the registers

kyle-kauzlarich replied:
I was hinting at the spacing convention

DenEscobar committed:
Passing test

DenEscobar committed:
Un-hardcoding program...
-ls8.py takes in a file parameter for load()
-Reading in file line by line in cpu.py and ignoring blank lines

DenEscobar committed:
load() will not take in any file and save instructions to ram

DenEscobar committed:
passig tests for running mult.ls8...
-changed load() to truely check for empty lines and lines that are comments
-appends integers instead of strings now
-moved assigments of aoperand variables within the while loop to keep them dynamic
-grabed values from register inorder to mulitply correctly

DenEscobar committed:
passing tests for day 3...
-added programs as variables for neater if statements
-added command instruction for push
-added command instruction for pop

DenEscobar committed:
added call and ret, passing tests
